reposerver/repository/repository.go

```diff
@@ -474,6 +474,7 @@ func resolveReferencedSources(hasMultipleSources bool, source *v1alpha1.Applicat
 					if !ok {
 						_, referencedCommitSHA, err := newClientResolveRevision(&refSourceMapping.Repo, refSourceMapping.TargetRevision)
 						if err != nil {
+							log.Errorf("failed to get git client for repo %s: %v", refSourceMapping.Repo.Repo, err)
 							return nil, fmt.Errorf("failed to get git client for repo %s", refSourceMapping.Repo.Repo)
```

Comment:
Should we add the error info to the returned error message as well?

Reply:
I think I intentionally avoided that since we don't know whether sensitive information might be in the error message. But maybe that's too paranoid.